packages/@aws-cdk/aws-apigateway/lib/deployment.ts

```diff
@@ -128,12 +128,34 @@ class LatestDeploymentResource extends CfnDeployment {
     if (this.hashComponents.length > 0) {
       const md5 = crypto.createHash('md5');
       this.hashComponents
-        .map(c => stack.resolve(c))
+        .map(c => {
+          try {
+            /*
+              stack.resolve() will error out when there is a cross stack reference to another stack. The entire stack
+              can be resolved only after completion of the 'prepare' phase. However, this method is part of the
+              preparation.
+              The correct thing to do here is to resolve the partially token within its current scope, as being done
+              in the catch block below.
+
+              Leaving the incorrect logic in the try block to not break existing CDK apps. Apps without cross-stack
+              reference for API Gateway resource would not have encountered this error; however, changing this logic
+              would change the LogicalId of the Deployment resource and indicate Cloudformation to replace the resource.
```

Comment:
Thinking about this a little further, why is replacement of the deployment resource an issue? We replace it automatically every time the API changes anyway... It shouldn't cause a service disruption.